Content.Server/Botany/Systems/PlantAnalyzerSystem.cs

```diff
@@ -0,0 +1,261 @@
+        if (comp.HarvestRepeat == HarvestType.NoRepeat) plantHarvestType = Loc.GetString("plant-analyzer-harvest-ephemeral");
+        if (comp.HarvestRepeat == HarvestType.SelfHarvest) plantHarvestType = Loc.GetString("plant-analyzer-harvest-autoharvest");
+
+        seedChem += String.Join("\n   ", comp.Chemicals.Select(item => item.Key.ToString()));
+
+        exudeGases = comp.ExudeGasses.Count > 0
+            ? string.Join("\n   ", comp.ExudeGasses.Keys)
+            : Loc.GetString("plant-analyzer-plant-gasses-no");
+
+        List<string> tolerancesList = CheckAllTolerances(comp);
+        List<string> generalTraitsList = CheckGeneralTraits(comp);
+        List<string> mutationsList = CheckAllMutations(comp);
+
+        return new PlantAnalyzerScannedSeedPlantInformation(GetNetEntity(target), seedName,
+            seedChem, plantHarvestType, exudeGases, mutationsList, false, tolerancesList, generalTraitsList, scanMode);
+    }
+
+    /// <summary>
+    ///     Analysis of unique seed.
+    /// </summary>
+    public PlantAnalyzerScannedSeedPlantInformation ObtainingGeneDataSeed(SeedData comp, EntityUid target, bool trayChecker, Boolean scanMode)
+    {
+        var seedName = Loc.GetString(comp.DisplayName);
+
+        var seedChem = "\n   ";
+        var plantHarvestType = "";
+        var exudeGases = "";
+
+        if (comp.HarvestRepeat == HarvestType.Repeat) plantHarvestType = Loc.GetString("plant-analyzer-harvest-repeat");
+        if (comp.HarvestRepeat == HarvestType.NoRepeat) plantHarvestType = Loc.GetString("plant-analyzer-harvest-ephemeral");
+        if (comp.HarvestRepeat == HarvestType.SelfHarvest) plantHarvestType = Loc.GetString("plant-analyzer-harvest-autoharvest");
+
+        seedChem += string.Join("\n   ", comp.Chemicals.Select(item => item.Key.ToString()));
+
+        exudeGases = comp.ExudeGasses.Count > 0
+            ? string.Join("\n   ", comp.ExudeGasses.Keys)
+            : Loc.GetString("plant-analyzer-plant-gasses-no");
+
+        List<string> tolerancesList = new List<string>();
+        List<string> mutationsList = new List<string>();
+
+        if (scanMode)
+        {
+            tolerancesList = CheckAllTolerances(comp);
+            mutationsList = CheckAllMutations(comp);
+        }
+
+        List<string> generalTraitsList = CheckGeneralTraits(comp);
+
+        return new PlantAnalyzerScannedSeedPlantInformation(GetNetEntity(target), seedName,
+            seedChem, plantHarvestType, exudeGases, mutationsList, trayChecker, tolerancesList, generalTraitsList, scanMode);
+    }
+
+    /// <summary>
+    ///     Returns information about the seed.
+    /// </summary>
+    public List<string> CheckGeneralTraits(SeedData plant)
+    {
+        var name = Loc.GetString(plant.DisplayName);
+        float endurance = plant.Endurance;
+        float yield = plant.Yield;
+        float lifespan = plant.Lifespan;
+        float maturation = plant.Maturation;
```

Comment:
Please don't use C# keywords for variable names.

Reply:
Makes sense, for the record, "Yield" is still used in SeedPrototype.cs